From 54e6171b6791efaf9d5bc8280d5810fad0a482e8 Mon Sep 17 00:00:00 2001
From: John Zhuge <jzhuge@cloudera.com>
Date: Wed, 3 May 2017 01:02:23 -0700
Subject: [PATCH 2399/2848] HDFS-11710. hadoop-hdfs-native-client build fails
 in trunk in Windows after HDFS-11529 (Contributed
 by Sailesh Mukil)

(cherry picked from commit b0f54ea035f406909f45c66b5403580919d63a4a)

Conflicts:
	hadoop-hdfs-project/hadoop-hdfs-native-client/src/main/native/libhdfs/os/windows/thread_local_storage.c

Change-Id: Ia14ed51e483d3d4d94829cf96a71b9b54410ca91
---
 .../libhdfs/os/windows/thread_local_storage.c      |    7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/native/libhdfs/os/windows/thread_local_storage.c b/hadoop-hdfs-project/hadoop-hdfs/src/main/native/libhdfs/os/windows/thread_local_storage.c
index 8583dc3..fc5002c 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/native/libhdfs/os/windows/thread_local_storage.c
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/native/libhdfs/os/windows/thread_local_storage.c
@@ -61,6 +61,13 @@ static void detachCurrentThreadFromJvm()
   free(state);
 }
 
+void hdfsThreadDestructor(void *v)
+{
+  // Ignore 'v' since it will contain the state and we will obtain it in the below
+  // call anyway.
+  detachCurrentThreadFromJvm();
+}
+
 /**
  * Unlike pthreads, the Windows API does not seem to provide a convenient way to
  * hook a callback onto thread shutdown.  However, the Windows portable
-- 
1.7.9.5

